frontends/p4/toP4/toP4.cpp

```diff
@@ -351,7 +351,10 @@ bool ToP4::preorder(const IR::Type_SerEnum* t) {
     dump(1);
     visit(t->annotations);
     builder.append("enum ");
-    visit(t->type);
+    if (t->type)
```

Comment:
This should not be necessary with the changes proposed below.

Reply:
It is necessary.  This is a note I added in a prior comment of this PR.

"Next set of changes relate to how `make check` tests each P4 program. The Python script for each test pretty prints (using the `PrettyPrint` pass) the P4 program first. `PrettyPrint` calls ToP4 to output the P4 program. However, ToP4 finds the serEnum->type to be null and fails to write the type for serEnum which is incorrect."